### PR TITLE
feat: add playwright e2e tests for DevGuard Web

### LINKED ISSUE #2079: E2E Testing of DevGuard Web with Playwright
Refactor and extend the existing Playwright setup to cover DevGuard Web usability testing, focusing on high-level, initial core functionalities rather than deeply nested edge cases or isolated UI components.

---

Update existing Playwright workflows and get them running

- [x] Login Flow
- [x] Light/Dark Mode
- [x] Repo Flow
- [x] Prerelease

New tests to implement

- [x] Organization Creation
- [x] Group Creation
- [x] Repository Creation
- [x] Invitation Flow (invite a user to an org/project)
- [x] Logout + LoginAgain (explicit logout, redirect to login page)
- [x] Setup Risk Scanning for GitLab Repo
- [x] Filter Vulns

Additional tests to implement

- [x] Manually start a DevGuard Scan using a SBOM
- [x] Handle a vulnerability after DevGuard Scan

Refactoring

- [x] Refactor POM to classes with dedicated functions

### PR DESCRIPTION
**Pull request by juliankepka**
https://github.com/l3montree-dev/devguard/issues/2079

### CODE REVIEW

File: e2e/src/pom/devguard.ts [hunk 1]
```diff
@@ -227,33 +221,30 @@ export class DevGuardPOM {
...
     await this.page.waitForTimeout(500);
     await this.page
-      .getByRole("heading", { name: "Use your own Scanner Expert" })
+      .getByTestId('own-setup-card')
       .click();
-    await this.page.locator('button[id="scanner-selection-continue"]').click();
+    await this.page.getByTestId('scanner-selection-continue').click();
   }
 
   async setupFlow_selectManualUpload() {
```
Comment (timbastin) [inline e2e/src/pom/devguard.ts]: What do you think of instead of prefixing those functions with `setupFlow` returning a new SetupFlow class instead which provides such functions.

So like

```
class DevGuardPOM {
   openVulnerabilityScanningSetup(): SetupFlow
}

...

class SetupFlow {
   selectManualUpload()
}
```

This does not only apply to the setup flow, but to a lot of functions of the devguard pom. I. think there is a lot of refactoring opportunity.

Comment (juliankepka) [reply e2e/src/pom/devguard.ts]: Great suggestion! Extracting sub-flows into dedicated classes fits with the POM philosophy. Each class represents a coherent piece of UI state, and the return type itself communicates intent. I'll try to refactor these functions.

File: e2e/src/pom/flows/setup.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    await this.page.waitForTimeout(500);
+    await this.page.getByTestId("manual-upload-card").click();
+    await this.page.getByTestId("integration-method-selection-continue").click();
+  }
+
+  async uploadSbomFile(inputFile: string) {
+    await this.page.waitForTimeout(500);
+    await this.page.getByTestId("sbom-tab").click();
+    await this.page.getByTestId("file-upload-input").setInputFiles(inputFile);
+    await this.page.getByTestId("manual-integration-continue").click();
```
[the remark this answers is not in the document]
Comment (juliankepka) [reply e2e/src/pom/flows/setup.ts:27]: Fixed through previous comment fix

File: e2e/src/pom/flows/auth.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    await expect(
+      this.page.getByRole("checkbox", { name: "I agree to the terms of use" }),
+    ).toBeChecked();
+
+    await this.page.getByRole("button", { name: "Sign up" }).click();
+    await this.page.getByTestId("password-button").click();
+    await this.page.getByTestId("password").fill(password);
+    await this.page.getByRole("button", { name: "Sign up", exact: true }).click();
+
+    console.log("Registered user with email: " + username + " and password: " + password);
```
[the remark this answers is not in the document]
Comment (juliankepka) [reply e2e/src/pom/flows/auth.ts]: Just a test case but changed to username and email instead of pw

File: e2e/src/utils.ts [hunk 4]
```diff
@@ -74,6 +74,10 @@ function loadEnvVariables() {
       domain: process.env.DEVGUARD_DOMAIN!,
       uniqueUsername: () =>
         config.devGuard.usernameTemplate.replace("XXX", Date.now().toString()),
+      uniqueEMail: () =>
+        config.devGuard.usernameTemplate
+          .replace("XXX", Date.now().toString())
+          .concat("@example.com"),
```
[the remark this answers is not in the document]
Comment (juliankepka) [reply e2e/src/utils.ts:86]: its just a test case and if this problem occurs i will check on it

File: e2e/src/devguard-repo-flow.e2e.spec.ts [hunk 5]
```diff
@@ -1,179 +1,42 @@
...
       "http://github.com/user-attachments/files/23216827/vex_l3montree_web_reopened.json",
     );
     await page.waitForTimeout(1_000);
-    await devguardPOM.deleteFirstArtifact();
-    await page.waitForTimeout(1_000); // wait until page has been updated (first item was removed)
-    await devguardPOM.deleteFirstArtifact();
+    await devguardPOM.artifacts().deleteFirstArtifact();
+    await page.waitForTimeout(1_000);
+    await devguardPOM.artifacts().deleteFirstArtifact();
     await page
```
[the remark this answers is not in the document]
Comment (juliankepka) [reply e2e/src/devguard-repo-flow.e2e.spec.ts:42]: The two deleteFirstArtifact() calls are intentional: setupSbomUpload() creates one artifact (the uploaded SBOM), and createNewArtifact() creates a second one. Both need to be deleted to reach the "No Artifacts Available" state that the assertion checks. The test is correct as-is.

**Review by seb-kw — CHANGES_REQUESTED** [review]
Please resolve conflicts again 😅

**Review by timbastin — APPROVED** [review]
